File: backend/services/aiService.js
const axios = require('axios')

class AIService {
  constructor() {
    this.provider = process.env.AI_PROVIDER || 'kimi'
    this.kimiKey = process.env.KIMI_API_KEY
    this.stepfunKey = process.env.STEPFUN_API_KEY
  }

  async generateContent(rawContent, category) {
    const categoryContext = this.getCategoryContext(category)
    
    const prompt = `
你是一位专业的技术内容编辑，请将以下原始内容转化为四层结构的深度解析。

${categoryContext}

原始内容：
${rawContent}

分类：${category}

请按以下结构生成内容（所有内容必须用中文，专业术语可以括号标注英文）：

1. 概念层（200-400字）：用通俗易懂的语言解释核心概念，让非专业人士也能理解
2. 原理层（2000-4000字）：深入讲解技术原理、工作机制、理论基础，包含数学模型或生物机制
3. 实现层（3000-6000字）：提供详细的技术实现、算法流程、代码示例或实验方法
4. 应用层（2000-3000字）：分析行业应用场景、实际案例、市场前景和未来发展

内容要求：
- 必须包含具体的数据、图表描述、对比分析
- 原理层要有数学公式或生物机制的详细说明
- 实现层要有伪代码或算法流程
- 应用层要有真实案例和市场分析
- 总字数不少于8000字

请以JSON格式返回：
{
  "title": "文章标题（中文）",
  "summary": "文章摘要（200-300字）",
  "layers": [
    {"level": "concept", "content": "概念层内容（使用markdown格式，包含标题、段落）"},
    {"level": "principle", "content": "原理层内容（使用markdown格式，包含公式、表格）"},
    {"level": "implementation", "content": "实现层内容（使用markdown格式，包含代码块）"},
    {"level": "application", "content": "应用层内容（使用markdown格式，包含案例分析）"}
  ],
  "tags": ["标签1", "标签2", "标签3"]
}
    `

    return await this.callAI(prompt)
  }

  getCategoryContext(category) {
    const contexts = {
      'world-model': '这是关于AI世界模型的内容。世界模型是AI系统理解和预测环境的核心能力，涉及表征学习、因果推理、时空建模等。请重点关注模型架构、训练方法、应用场景。',
      'embodied': '这是关于具身智能的内容。具身智能强调AI与物理世界的交互，包括机器人控制、感知-行动循环、多模态学习。请重点关注硬件集成、控制算法、实际部署。',
      'bci': '这是关于脑机接口的内容。脑机接口连接大脑与外部设备，涉及神经信号采集、解码算法、应用场景。请重点关注信号处理、机器学习方法、医疗应用。',
      'chip': '这是关于芯片架构的内容。芯片架构决定计算性能和能效，涉及处理器设计、存储层次、互连网络。请重点关注架构创新、性能优化、制造工艺。',
      'biotech': '这是关于生物医疗的内容。生物医疗结合生物学和技术创新，涉及基因编辑、药物研发、精准医疗。请重点关注技术原理、临床应用、市场前景和监管挑战。'
    }
    return contexts[category] || '这是前沿技术内容，请深入分析其技术原理和应用价值。'
  }

  async generateAIResponse(context, history = []) {
    const messages = [
      { role: 'system', content: '你是一位专业的AI技术助手，帮助用户理解复杂的技术概念。' },
      ...history,
      { role: 'user', content: context }
    ]

    return await this.callAI(messages)
  }

  async callAI(prompt) {
    try {
      if (this.provider === 'kimi') {
        return await this.callKimi(prompt)
      } else if (this.provider === 'stepfun') {
        return await this.callStepFun(prompt)
      }
    } catch (error) {
      console.error('AI调用失败:', error.response?.data || error.message)
      throw new Error('AI服务暂时不可用: ' + (error.response?.data?.error?.message || error.message))
    }
  }

  async callKimi(prompt) {
    const response = await axios.post(
      'https://api.moonshot.cn/v1/chat/completions',
      {
        model: 'moonshot-v1-32k',
        messages: typeof prompt === 'string' 
          ? [{ role: 'user', content: prompt }]
          : prompt,
        temperature: 0.7
      },
      {
        headers: {
          'Authorization': `Bearer ${this.kimiKey}`,
          'Content-Type': 'application/json'
        }
      }
    )
    
    return response.data.choices[0].message.content
  }

  async callStepFun(prompt) {
    const response = await axios.post(
      'https://api.stepfun.com/v1/chat/completions',
      {
        model: 'step-1-32k',
        messages: typeof prompt === 'string'
          ? [{ role: 'user', content: prompt }]
          : prompt,
        temperature: 0.7
      },
      {
        headers: {
          'Authorization': `Bearer ${this.stepfunKey}`,
          'Content-Type': 'application/json'
        }
      }
    )
    
    return response.data.choices[0].message.content
  }
}

module.exports = new AIService()

File: backend/services/ai_service.py
import os
import requests
from typing import Optional, Dict
import json
from .news_service import NewsService

class AIService:
    """AI模型服务 - 支持阶跃星辰和Kimi，集成实时新闻"""
    
    def __init__(self):
        # 阶跃星辰配置
        self.stepfun_api_key = os.getenv('STEPFUN_API_KEY', '')
        self.stepfun_url = 'https://api.stepfun.com/v1/chat/completions'
        # 使用Step 1v 8k模型（更经济的选择）
        self.stepfun_model = os.getenv('STEPFUN_MODEL', 'step-1v-8k')
        
        # Kimi配置
        self.kimi_api_key = os.getenv('KIMI_API_KEY', '')
        self.kimi_url = 'https://api.moonshot.cn/v1/chat/completions'
        
        # 新闻服务
        self.news_service = NewsService()
        
        # 默认使用阶跃星辰
        self.current_provider = 'stepfun'
        
        print(f"✅ AI服务初始化: 使用阶跃星辰模型 {self.stepfun_model}")
    
    def analyze_sentiment(self, 
                         stock_symbol: str, 
                         stock_name: str,
                         system_prompt: Optional[str] = None,
                         provider: str = 'stepfun',
                         include_news: bool = True) -> str:
        """
        分析股票舆情
        
        Args:
            stock_symbol: 股票代码
            stock_name: 股票名称
            system_prompt: 自定义系统提示词
            provider: AI提供商 ('stepfun' 或 'kimi')
            include_news: 是否包含实时新闻
        """
        if system_prompt is None:
            system_prompt = """你是一个专业的美股市场分析师。请基于提供的最新新闻和市场动态，
为指定股票提供简短的舆情分析（不超过30字）。
重点关注：
1. 近期重大新闻事件
2. 市场情绪变化
3. 短期价格影响因素
请用简洁、专业的语言回答。今天是2026年3月10日。"""
        
        # 获取实时新闻
        news_context = ""
        if include_news:
            try:
                news_list = self.news_service.get_stock_news(stock_symbol, limit=3)
                if news_list:
                    news_context = self.news_service.format_news_for_ai(news_list)
                else:
                    news_context = "暂无该股票的最新新闻。"
            except Exception as e:
                print(f"获取新闻失败: {e}")
                news_context = "新闻获取失败，基于一般市场情况分析。"
        
        user_prompt = f"""请分析 {stock_name}({stock_symbol}) 的当前舆情和短期影响。

{news_context}

请基于以上最新信息，用不超过30字简要分析该股票的舆情和短期影响。"""
        
        try:
            if provider == 'stepfun':
                return self._call_stepfun(system_prompt, user_prompt)
            elif provider == 'kimi':
                return self._call_kimi(system_prompt, user_prompt)
            else:
                return "不支持的AI提供商"
        except Exception as e:
            print(f"AI分析失败: {e}")
            return "暂无舆情数据"
    
    def _call_stepfun(self, system_prompt: str, user_prompt: str, max_tokens: int = 100, model: str = None) -> str:
        """
        调用阶跃星辰API
        
        Args:
            system_prompt: 系统提示词
            user_prompt: 用户提示词
            max_tokens: 最大token数
            model: 指定模型（如果为None则使用默认模型）
        """
        # 如果没有指定模型，使用默认的8k模型
        use_model = model if model else self.stepfun_model
        
        headers = {
            'Authorization': f'Bearer {self.stepfun_api_key}',
            'Content-Type': 'application/json'
        }
        
        data = {
            'model': use_model,
            'messages': [
                {'role': 'system', 'content': system_prompt},
                {'role': 'user', 'content': user_prompt}
            ],
            'temperature': 0.7,
            'max_tokens': max_tokens
        }
        
        print(f"🤖 调用阶跃星辰API: 模型={use_model}, max_tokens={max_tokens}")
        
        try:
            response = requests.post(self.stepfun_url, headers=headers, json=data, timeout=30)
            
            if response.status_code == 200:
                result = response.json()
                content = result['choices'][0]['message']['content'].strip()
                print(f"✅ AI响应成功: {len(content)} 字符")
                return content
            else:
                error_msg = f"阶跃星辰API错误: {response.status_code} - {response.text}"
                print(f"❌ {error_msg}")
                return f"AI分析暂时不可用 (错误码: {response.status_code})"
        except requests.exceptions.Timeout:
            print("❌ API请求超时")
            return "AI分析超时，请稍后重试"
        except Exception as e:
            print(f"❌ API调用异常: {type(e).__name__}: {e}")
            return f"AI分析失败: {str(e)}"
    
    def _call_kimi(self, system_prompt: str, user_prompt: str) -> str:
        """调用Kimi API"""
        headers = {
            'Authorization': f'Bearer {self.kimi_api_key}',
            'Content-Type': 'application/json'
        }
        
        data = {
            'model': 'moonshot-v1-8k',
            'messages': [
                {'role': 'system', 'content': system_prompt},
                {'role': 'user', 'content': user_prompt}
            ],
            'temperature': 0.7,
            'max_tokens': 100
        }
        
        response = requests.post(self.kimi_url, headers=headers, json=data, timeout=30)
        
        if response.status_code == 200:
            result = response.json()
            return result['choices'][0]['message']['content'].strip()
        else:
            print(f"Kimi API错误: {response.status_code} - {response.text}")
            return "AI分析暂时不可用"
    
    def batch_analyze_sentiment(self, 
                                stocks: list,
                                system_prompt: Optional[str] = None,
                                provider: str = 'stepfun') -> Dict[str, str]:
        """批量分析股票舆情"""
        results = {}
        for stock in stocks:
            symbol = stock.get('symbol')
            name = stock.get('name', symbol)
            sentiment = self.analyze_sentiment(symbol, name, system_prompt, provider)
            results[symbol] = sentiment
        return results
